### PR TITLE
Fwmt 973 pass update pause request to comet

### PR DESCRIPTION
**Pull request by SteveScorfield**

### CODE REVIEW

File: src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java [hunk 1]
```diff
@@ -41,5 +45,15 @@ public void sendJobRequest(ActionInstruction actionInstruction) throws GatewayEx
...
       }
     }
+    else if (actionInstruction.getActionUpdate() != null){
+      if (!StringUtils.isEmpty(actionInstruction.getActionUpdate().getPause().getId())) {
+        jobServiceProducer.sendMessage(CanonicalJobHelper.newUpdateJob(actionInstruction));
+        gatewayEventManager
+            .triggerEvent(actionInstruction.getActionUpdate().getPause().getId(), CANONICAL_UPDATE_SENT, LocalTime.now());
+      } else {
+        gatewayEventManager
+            .triggerEvent("00000000-0000-0000-0000-000000000000", CANONICAL_UPDATE_FAILED, LocalTime.now());
```
Comment (JacobHarrisonONS) [inline src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java]: what are you doing here?

File: src/main/resources/xsd/actionInstruction.xsd [hunk 2]
```diff
@@ -127,7 +127,8 @@
             <xs:element name="effectiveDate" type="xs:date" minOccurs="0" maxOccurs="1"/>
             <xs:element name="code" type="xs:string" minOccurs="0" maxOccurs="1"/>
             <xs:element name="reason" type="xs:string" minOccurs="0" maxOccurs="1"/>
-            <xs:element name="holdUntil" type="xs:date" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="until" type="xs:date" minOccurs="0" maxOccurs="1"/>
```
Comment (JacobHarrisonONS) [inline src/main/resources/xsd/actionInstruction.xsd:130]: these changes are from datamap?

File: src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java [hunk 3]
```diff
@@ -41,5 +45,15 @@ public void sendJobRequest(ActionInstruction actionInstruction) throws GatewayEx
             .triggerEvent(actionInstruction.getActionCancel().getCaseId(), CANONICAL_CANCEL_FAILED, LocalTime.now());
       }
     }
+    else if (actionInstruction.getActionUpdate() != null){
+      if (!StringUtils.isEmpty(actionInstruction.getActionUpdate().getPause().getId())) {
+        jobServiceProducer.sendMessage(CanonicalJobHelper.newUpdateJob(actionInstruction));
+        gatewayEventManager
+            .triggerEvent(actionInstruction.getActionUpdate().getPause().getId(), CANONICAL_UPDATE_SENT, LocalTime.now());
```
Comment (JacobHarrisonONS) [inline src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java]: if its supposed to contain an id but doesn't throw exception, if it cannot match request to any of the types should also throw an exception

File: src/main/java/uk/gov/ons/census/fwmt/rmadapter/canonical/CanonicalJobHelper.java [hunk 4]
```diff
@@ -190,9 +191,29 @@ private static void createIndefinitePause(CancelFieldWorkerJobRequest cancelJobR
...
+
+      updateJobRequest.setActionType("update");
+
+      if (!StringUtils.isEmpty(actionPause.getUntil())) {
+        dateConversion = OffsetDateTime.parse(actionPause.getUntil().toString());
+      } else {
+        dateConversion = OffsetDateTime.now().plusDays(7);
+      }
+
+      updateJobRequest.setUntil(dateConversion);
```
Comment (JacobHarrisonONS) [inline src/main/java/uk/gov/ons/census/fwmt/rmadapter/canonical/CanonicalJobHelper.java:211]: sends null update if pause is empty

**Review by JacobHarrisonONS — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java (hunk 1 above)
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java]: This has been updated as discussed earlier

File: src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java (hunk 3 above)
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java]: Updated as discussed

File: src/main/java/uk/gov/ons/census/fwmt/rmadapter/canonical/CanonicalJobHelper.java (hunk 4 above)
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/census/fwmt/rmadapter/canonical/CanonicalJobHelper.java:211]: Updated as discussed

**Review by JacobHarrisonONS — DISMISSED** [review]
looks good

**Review by JacobHarrisonONS — APPROVED** [review]
lgtm

**Pull request merged**